pkg/tests/ossm/smcp_must_gather_test.go

```diff
@@ -40,18 +45,171 @@ func TestMustGather(t *testing.T) {
+			t.LogStep("verify version exist")
+			fileList := []string{"version"}
+			verifyFilesExist(t, dir, fileList)
+
+			t.LogStep("verify file version contains the version of the must-gather image")
+			versionFilePath := getPathToFile(t, dir, "version")
+			shell.Execute(t,
+				fmt.Sprintf("cat %s", versionFilePath),
+				assert.OutputContains(env.GetMustGatherTag(),
+					"Expected must gather version was found",
+					"Expected must gather version was not found"))
+		})
+
+		t.NewSubTest("resource cluster scoped exist").Run(func(t TestHelper) {
+			t.LogStep("Verify resources for cluster scoped files are created")
+			t.Log("verify that resources for cluster scoped are created: nodes, clusterrolebindings, clusterroles")
+			clusterScopedResourcesPathList := []string{
+				"cluster-scoped-resources/core/nodes/*.yaml",
+				"cluster-scoped-resources/rbac.authorization.k8s.io/clusterrolebindings/*.yaml",
+				"cluster-scoped-resources/rbac.authorization.k8s.io/clusterroles/*.yaml"}
+			for _, path := range clusterScopedResourcesPathList {
+				pathAndFilesExist(t, dir, path)
+			}
+		})
+
+		t.NewSubTest("resource for namespaces exist").Run(func(t TestHelper) {
+			t.LogStep("verify that resources for namespaces are created including bookinfo and istio-system folders")
+			namespacedResourcesPathList := []string{
+				"namespaces/istio-system/*.yaml",
+				"namespaces/bookinfo/*.yaml",
+				"namespaces/openshift-operators/openshift-operators.yaml",
+				"namespaces/*/k8s.cni.cncf.io/network-attachment-definitions/*.yaml",
+				"namespaces/*/rbac.authorization.k8s.io/rolebindings/*.yaml"}
+			for _, path := range namespacedResourcesPathList {
+				pathAndFilesExist(t, dir, path)
+			}
+		})
+
+		t.NewSubTest("cluster service version files validation").Run(func(t TestHelper) {
+			t.LogStep("Get service current service version from the cluster")
+			csvList := shell.Execute(t, "oc get csv -n openshift-operators | awk 'NR>1 { print $1 }'")
+
+			t.LogStep("verify if the csv files exist for the current service version")
+			csvListSlice := strings.Split(csvList, "\n")
+			for _, csv := range csvListSlice {
+				if csv != "" {
+					pathAndFilesExist(t,
+						dir,
+						fmt.Sprintf("namespaces/openshift-operators/operators.coreos.com/clusterserviceversions/%s.yaml", csv))
+				}
+			}
+		})
 	})
 }
+
+func verifyFilesExist(t TestHelper, dir string, fileList []string) {
+	fileMatches := make(map[string]bool)
+
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		for _, file := range fileList {
+			match, _ := filepath.Match(file, info.Name())
+			if match {
+				fileMatches[file] = true
+			}
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		t.Fatalf("Error: %s\n", err.Error())
+	}
+
+	for _, file := range fileList {
+		if !fileMatches[file] {
+			t.Fatalf("File not found: %s", file)
+		}
+	}
+}
+
+func pathAndFilesExist(t TestHelper, dir string, path string) {
```

Comment:
This looks very similar to `verifyFilesExist()`. Why do you need two functions for (virtually) the same thing?